Objects/regions.c

```diff
@@ -88,6 +89,168 @@ bool is_c_wrapper(PyObject* obj){
     return PyCFunction_Check(obj) || Py_IS_TYPE(obj, &_PyMethodWrapper_Type) || Py_IS_TYPE(obj, &PyWrapperDescr_Type);
 }
 
+/**
+ * Global status for performing the region check.
+ */
+bool do_region_check = false;
+
+// The src object for an edge that invalidated the invariant.
+PyObject* error_src = Py_None;
+
+// The tgt object for an edge that invalidated the invariant.
+PyObject* error_tgt = Py_None;
+
+// Once an error has occurred this is used to surpress further checking
+bool error_occurred = false;
+
+
+/**
+ * Enable the region check.
+ */
+void notify_regions_in_use(void)
+{
+    // Do not re-enable, if we have detected a fault.
+    if (!error_occurred)
+        do_region_check = true;
+}
+
+PyObject* _Py_EnableInvariant(void)
+{
+    // Disable failure as program has explicitly requested invariant to be checked again.
+    error_occurred = false;
+    // Re-enable region check
+    do_region_check = true;
+    return Py_None;
+}
+
+/**
+ * Set the global variables for a failure.
+ * This allows the interpreter to inspect what has failed.
+ */
+void set_failed_edge(PyObject* src, PyObject* tgt)
+{
+    Py_IncRef(src);
+    error_src = src;
+    Py_IncRef(tgt);
+    error_tgt = tgt;
```

Comment:
Here we should have a check that the values are currently `Py_None` or call `Py_DecRef()` on the nodes before overwriting them

Other option would be to have the `Py_DecRef` in `_Py_EnableInvariant` and set them to `Py_None` again